README.md

```diff
@@ -1,23 +1,133 @@
+
+```java
+@GetMapping("/users/{userId}") 
+public User getUser(@PathVariable UUID userId) {
+        [implementation deleted]
+    }
+```
+
+- Without using the Friendly ID library, you could access a given user as follows:
+`curl http://localhost:8080/users/c3587ec5-0976-497f-8374-61e0c2ea3da5`
+
+- After adding the FriendlyID library, the controller method itself does not change, but you would be able to access a given user using the relevant FriendlyID as follows:
+`curl http://localhost:8080/users/5wbwf6yUxVBcr48AMbz9cb`
+
+In addition, if a given document returned by such a method contains objects of type UUID, those IDs will also be shortened into FriendlyID format.
+
+
+2. Advanced (Optimizing testing)
+
+ 
+ The FriendlyID library makes it possible to define for UUIDs values which are easy to read. By using names instead of hard-to-remember UUIDs, you can write much simpler tests for your code, for example:
+ 
+```java
+@Test 
+public void shouldGetUser() { 
+mockMvc.perform(get("/users/{userId}", "John")) 
+.andExpect(status().isOk()) 
+.andExpect(content().contentType("application/json")) 
+.andExpect(jsonPath("$.uuid", is("John"))); 
+} 
+```
+
+In the above example, the variable "John" is decoded by the library to the correct UUID, in this case, `00000000-0000-0000-0000-000000a69efb`. In this way, you can give a variable in a test class a truly meaningful value and, as a result, an assertion which refers to that variable becomes exceptionally easy to understand in your test program.
+
+
+FriendlyID library 
+--
+
+Dependencies
+---
+
+```xml
+<dependency>
+    <groupId>com.devskiller.friendly-id</groupId>
+    <artifactId>friendly-id</artifactId>
+    <version>1.1.0</version>
+</dependency>
+```
+
+Usage
 ---
 
-Just add friendly-id starter dependency:
+1. ```java
+FriendlyId.createFriendlyId();
+```
+
+This creates a new, random FriendlyID, for example: `5wbwf6yUxVBcr48AMbz9cb`
+
+
+2. ```java
+FriendlyId.toFriendlyId(UUID.fromString("c3587ec5-0976-497f-8374-61e0c2ea3da5"));
+```
+
+This converts a UUID in the form of a string to a FriendlyID, for example: `5wbwf6yUxVBcr48AMbz9cb`
+
+3. ```java
+FriendlyId.toUuid("5wbwf6yUxVBcr48AMbz9cb");
+```
+
+This converts a FriendlyID to its UUID, for example: `c3587ec5-0976-497f-8374-61e0c2ea3da5`
+
+
+Notes
+--
+	
+* As every *UUID* is a 128-bit number, a *FriendlyID* can also store only a 128-bit number.
+* If a FriendlyID has any leading zeros, those leading zeros are ignored - for example, `00cafe` is treated as `cafe`.
+
+
+Integrations
+--
+
+- [Spring Boot integration](#Spring-Boot-integration)
+- [Jackson integration ](#Jackson-integration)
+
+Spring Boot integration
+---
+1. The FriendlyID library includes a Spring configuration to make it easy to add shorter IDs to an application. With a typical application based on Spring Boot, for your controllers to be able to use FriendlyIDs when communicating with the outside world, just add one new starter dependency as follows:
```

Comment:
It could be helpful to name the autoconfiguration class in this section (in case someone would like to exclude/include it in their tests).